x/staking/keeper/grpc_query.go

```diff
@@ -180,19 +180,25 @@ func (k Querier) ValidatorUnbondingDelegations(ctx context.Context, req *types.Q
 	}
 
 	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
-	srcValPrefix := types.GetUBDsByValIndexKey(valAddr)
-	ubdStore := prefix.NewStore(store, srcValPrefix)
-	pageRes, err := query.Paginate(ubdStore, req.Pagination, func(key, value []byte) error {
-		storeKey := types.GetUBDKeyFromValIndexKey(append(srcValPrefix, key...))
-		storeValue := store.Get(storeKey)
-
-		ubd, err := types.UnmarshalUBD(k.cdc, storeValue)
-		if err != nil {
-			return err
-		}
-		ubds = append(ubds, ubd)
-		return nil
-	})
+	_, pageRes, err := query.CollectionPaginate(
```

Comment:
I wouldn't collect the unbonding delegations in the closure here, I would just collect the keys first as collections.Pair[sdk.ValAddress, sdk.AccAddress], then after I would loop over the keys to fetch the unbonding delegations, I think this makes the code cleaner and assumes less thing on the underlying store implementation.